voyager/modules/client/cometbls/src/main.rs

```diff
@@ -308,10 +310,20 @@ impl ClientModuleServer<ModuleData, ModuleCall, ModuleCallback> for ModuleServer
                     None::<()>,
                 )
             })
-            .map(|header| match self.ctx.ibc_interface {
-                SupportedIbcInterface::IbcSolidity => header.encode_as::<EthAbi>(),
-                SupportedIbcInterface::IbcMoveAptos => header.encode_as::<Bcs>(),
-            })
+            .map(|mut header| match self.ctx.ibc_interface {
+                SupportedIbcInterface::IbcSolidity => Ok(header.encode_as::<EthAbi>()),
+                SupportedIbcInterface::IbcMoveAptos => {
+                    header.zero_knowledge_proof =
+                        reencode_zkp_for_move(&header.zero_knowledge_proof).map_err(|e| {
+                            ErrorObject::owned(
+                                FATAL_JSONRPC_ERROR_CODE,
+                                format!("unable to decode zkp: {}", e),
```

Comment:
decode? or reencode?

Reply:
its reencode bc we decode then reencode it using ark-bn254